uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Mail/BaseRequestNotificationTemplate.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_Service_LeaveRequestComment as LeaveRequestCommentService;
+
+abstract class CRM_HRLeaveAndAbsences_Mail_BaseRequestNotificationTemplate {
+
+  /**
+   * @var \CRM_HRLeaveAndAbsences_Service_LeaveRequestComment
+   */
+  private $leaveRequestCommentService;
+
+  /**
+   * @var array|null
+   *   Stores the list of option values for the LeaveRequest status_id field.
+   */
+  private $leaveStatuses;
+
+  /**
+   * @var array|null
+   *   Stores the list of option values for the LeaveRequest from_date_type field.
+   */
+  private $leaveRequestDayTypes;
+
+  /**
+   * CRM_HRLeaveAndAbsences_Mail_BaseRequestNotificationTemplate constructor.
+   *
+   * @param \CRM_HRLeaveAndAbsences_Service_LeaveRequestComment $leaveRequestComment
+   */
+  public function __construct(LeaveRequestCommentService $leaveRequestComment) {
+    $this->leaveRequestCommentService = $leaveRequestComment;
+  }
+
+  /**
+   * Gets the message template for this Template Type.
+   *
+   * @return array
+   *  An array containing the message template attributes/values
+   */
+  abstract public function getTemplate();
```

Comment:
I initially named it getTemplateID() but then I found it difficult to test the method in the child template classes since the ID is not fixed but just an auto increment ID in the database. 
I need it to return all the other details of the template so that the test can use that to ascertain the right template was returned. 
What about changing the name to getTemplateDetails?